apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -49,33 +70,84 @@ public void close() throws Exception {
     }
   }
 
-  public int writeBatch(Collection<RdaChange<TClaim>> allClaims) throws ProcessingException {
-    final Collection<RdaChange<TClaim>> claims = dedupChanges(allClaims);
+  /**
+   * Queries the RdaApiProgress table to get the maximum sequence number for our claim type.
+   *
+   * @return max sequence number or empty if there is no record for this claim type
+   * @throws ProcessingException if the operation fails
+   */
+  @Override
+  public Optional<Long> readMaxExistingSequenceNumber() throws ProcessingException {
+    try {
+      logger.info("running query to find max sequence number");
+      String query =
+          String.format(
+              "select p.%s from RdaApiProgress p where p.claimType='%s'",
+              RdaApiProgress.Fields.lastSequenceNumber, claimType.name());
```

Comment:
any reason to not using a parameter for the claimType?

Reply:
Yes because concrete classes that derive from this are dedicated to a single claim type and the user of those classes does not need to know or care what the actual claim type is.

Reply:
I meant at the query level.
```
      String query =
          String.format(
              "select p.%s from RdaApiProgress p where p.claimType=:claimType",
              RdaApiProgress.Fields.lastSequenceNumber);
      final List<Long> sequenceNumber =
          entityManager.createQuery(query, Long.class)
             .setParameter("claimType", claimType.name())
             .getResultList();
```
maybe I'm over thinking it

Reply:
Ah I see.  Good point.  I think given this string is from a constant it's not a big deal to just add it directly.